### ISSUE TITLE
Live reports preview

### ISSUE THREAD

**Issue by alexgauvin** [invoiceninja/ui#1299]
Would be nice to have a real live report preview in React UI.

**Comment by beganovich** [comment]
Hey, thanks for reaching to us. Would you be keen to explain a tiny bit more about the live preview? We feature live preview both for designs and resource previvews (invoices, etc). Thanks!

**Comment by turbo124** [comment]
I've just tagged a new release of the app, this should also contain the live previews for reports now :)

**Comment by alexgauvin** [comment]
@turbo124  I am on latest build and it's not rendering anything. 
Actually, it stays on processing mode forever.
![image](https://github.com/invoiceninja/ui/assets/5897903/3d4b8b04-b1cc-40fa-89f4-3fbcb87de9d9)

**Comment by turbo124** [comment]
@alexgauvin 

can you try and reduce the range down to say a month?

Also if you open the browser console, do you see some automatic requests occuring every couple of seconds?

**Comment by alexgauvin** [comment]
@turbo124 : Reducing to 7 days finally output some data ... after 7 min !!!

Here's a screenshot of console during process
![Screenshot_2](https://github.com/invoiceninja/ui/assets/5897903/8376fa56-1d59-4f7b-a7cc-af6f2f99bdf1)

**Comment by turbo124** [comment]
7 minutes does not sound right, i think the system would time out and stop polling well before this. Can you confirm?

**Comment by alexgauvin** [comment]
It does not time out, but it does not display as well

**Comment by alexgauvin** [comment]
Hi,

I'm still having issues with reports in React UI. Nothing ever display and Processing spinner goes round forever.
There are a bunch of errors in the browser's console. 409 errors, never seen these !!! Don't know if its related.
![Capture d’écran 2024-05-28 053202](https://github.com/invoiceninja/ui/assets/5897903/add7a29c-c939-4016-8a25-3aadd5eb3262)

And a drill down with more details on 409 error
![Capture d’écran 2024-05-28 053931](https://github.com/invoiceninja/ui/assets/5897903/893ca6d8-9eb9-439d-9aab-aa573a1e3334)


Nothing in the laravel.log file.

Thanks for your support

**Comment by beganovich** [comment]
409 means the resource isn't ready.

What does your setup look like? Is it an official Docker image or a manual installation?

**Comment by alexgauvin** [comment]
Manual installation on shared hosting

**Comment by turbo124** [comment]
@alexgauvin i am guessing here that the job may only be attempted every 5 minutes, rather than on a proper queue system where the job would run almost instantly. Hence the timeouts.

**Comment by turbo124** [comment]
@alexgauvin 

yes, the preview relies on a job being spun up to process the report data, the 409's you see in the browser is the front end polling for when this has been completed. when it has completed, the front end retrieves the data and displays it.

because of the way shared hosting works, i do not think it is possible to get the live previews.

**Comment by alexgauvin** [comment]
Mmm, even if I set cronjob to run every minute ?
If not, that means that React UI won't be able to serve reports on any shared hosting.
That's sad. It was actually working in v4.

**Comment by beganovich** [comment]
@turbo124 can we conditionally dispatch with sync if .env is toggled (for live previews)?

**Comment by turbo124** [comment]
@beganovich this would impact both Previews and Regular "Exports" as they both dispatch jobs, can we get confirmation that this also impacts the Export button?

**Comment by beganovich** [comment]
> [@beganovich](https://github.com/beganovich) this would impact both Previews and Regular "Exports" as they both dispatch jobs, can we get confirmation that this also impacts the Export button?

That's true, we use the same logic. Perhaps we can go with something like this.

```php
if ($request->get('action') === 'preview' && config('ninja.features.preview_reports_using_sync')) {
    PreviewReport::dispatchSync($user->company(), $request->all(), InvoiceExport::class, $hash);
} else {
    PreviewReport::dispatch($user->company(), $request->all(), InvoiceExport::class, $hash);
}
```
